sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysisTool.java

```diff
@@ -195,63 +193,77 @@ public void run(CommandLine line, ToolRunningContext context) throws Exception {
         }
 
         List<Contingency> contingencies = line.hasOption(CONTINGENCIES_FILE_OPTION)
-                ? ContingencyList.load(context.getFileSystem().getPath(line.getOptionValue(CONTINGENCIES_FILE_OPTION))).getContingencies(network)
-                : Collections.emptyList();
+            ? ContingencyList.load(context.getFileSystem().getPath(line.getOptionValue(CONTINGENCIES_FILE_OPTION))).getContingencies(network)
+            : Collections.emptyList();
 
-        List<SensitivityVariableSet> variableSets;
+        List<SensitivityVariableSet> variableSets = Collections.emptyList();
         if (line.hasOption(VARIABLE_SETS_FILE_OPTION)) {
             try (Reader reader = Files.newBufferedReader(context.getFileSystem().getPath(line.getOptionValue(VARIABLE_SETS_FILE_OPTION)), StandardCharsets.UTF_8)) {
                 variableSets = objectMapper.readValue(reader, new TypeReference<>() {
                 });
             }
-        } else {
-            variableSets = Collections.emptyList();
         }
 
         SensitivityFactorJsonReader factorsReader = new SensitivityFactorJsonReader(factorsFile);
 
         context.getOutputStream().println("Running analysis...");
         Stopwatch stopwatch = Stopwatch.createStarted();
         try (ComputationManager computationManager = DefaultComputationManagerConfig.load().createLongTimeExecutionComputationManager()) {
+            SensitivityAnalysisParametersRecord parametersRecord = new SensitivityAnalysisParametersRecord(factorsReader, params, network, contingencies,
+                variableSets, computationManager, outputFile, outputFileStatus, csv);
+            run(line, parametersRecord);
+        }
+        context.getOutputStream().println("Analysis done in " + stopwatch.elapsed(TimeUnit.MILLISECONDS) + " ms");
+    }
 
-            if (line.hasOption(SINGLE_OUTPUT)) {
-                if (csv) {
-                    throw new PowsyblException("Unsupported " + SINGLE_OUTPUT + " option does not support csv file as argument of " + OUTPUT_FILE_OPTION + ". Must be json.");
+    private record SensitivityAnalysisParametersRecord(SensitivityFactorJsonReader factorsReader,
+                                                       SensitivityAnalysisParameters params,
+                                                       Network network,
+                                                       List<Contingency> contingencies,
+                                                       List<SensitivityVariableSet> variableSets,
+                                                       ComputationManager computationManager,
+                                                       Path outputFile,
+                                                       Path outputFileStatus,
+                                                       boolean csv) {
+    }
+
+    private void run(CommandLine line, SensitivityAnalysisParametersRecord parametersRecord) throws IOException, PowsyblException {
```

Comment:
to be fixed. PowsyblException is a RuntimeException so shouldn't be here. IOException is sometimes catched sometimes not.

propose that all IOException-s are turned to UncheckedIOExceptions in the method, and then remove both throws declarations here.